pallets/funding/src/functions.rs

```diff
@@ -1309,7 +1332,20 @@ impl<T: Config> Pallet<T> {
 			InvestorType::Professional => project_metadata.contributing_ticket_sizes.professional,
 			InvestorType::Retail => project_metadata.contributing_ticket_sizes.retail,
 		};
+		let max_multiplier = match investor_type {
+			InvestorType::Retail => {
+				RetailParticipations::<T>::mutate(&did, |project_participations| {
+					if project_participations.contains(&project_id).not() {
+						// We don't care if it fails, since it means the user already has access to the max multiplier
+						let _ = project_participations.try_push(project_id);
+					}
+					retail_max_multiplier_for_participations(project_participations.len() as u8)
+				})
```

Comment:
nit: maybe a [`try_mutate`](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageMap.html#method.try_mutate) where we modify the `RetailParticipations` is more suitable?

Reply:
no, because this function always need to return a value, and try_mutate expects a `None` if I want to not change the value. So I would need to always return `Some` which would do the same as `mutate`